mls-rs/src/group/state.rs

```diff
@@ -2,25 +2,52 @@
 // Copyright by contributors to this project.
 // SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+use mls_rs_core::group::Member;
+
 use super::{
-    confirmation_tag::ConfirmationTag, proposal::ReInitProposal,
+    confirmation_tag::ConfirmationTag, member_from_leaf_node, proposal::ReInitProposal,
     transcript_hash::InterimTranscriptHash,
 };
-use crate::group::{GroupContext, TreeKemPublic};
+use crate::{
+    group::{GroupContext, TreeKemPublic},
+    tree_kem::node::LeafIndex,
+};
 
+#[cfg_attr(
+    all(feature = "ffi", not(test)),
+    safer_ffi_gen::ffi_type(clone, opaque)
+)]
 #[derive(Clone, Debug, PartialEq)]
 #[non_exhaustive]
 pub struct GroupState {
     #[cfg(feature = "by_ref_proposal")]
     pub(crate) proposals: crate::group::ProposalCache,
-    pub(crate) context: GroupContext,
+    pub context: GroupContext,
     pub(crate) public_tree: TreeKemPublic,
     pub(crate) interim_transcript_hash: InterimTranscriptHash,
     pub(crate) pending_reinit: Option<ReInitProposal>,
```

Comment:
Should this be accessible to applications too?

Reply:
I eventually want to make mostly all of this public but it felt like that should be a separate PR later. Since those changes are only additive it could be done in a patch release.